extensions/amp-video/1.0/storybook/Basic.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {VideoWrapper} from '../video-wrapper';
+import {boolean, object, text, withKnobs} from '@storybook/addon-knobs';
+import {withA11y} from '@storybook/addon-a11y';
+
+const {forwardRef, useRef, useImperativeHandle} = Preact;
+
+export default {
+  title: 'Video Wrapper',
+  component: VideoWrapper,
+  decorators: [withA11y, withKnobs],
+};
+
+const VideoTagPlayer = forwardRef((props, ref) => {
+  const {
+    children,
+
+    src,
+    muted,
+    loop,
+    controls,
+    poster,
+
+    'aria-label': ariaLabel,
+    album,
+    artist,
+    artwork,
+    title,
+
+    onLoad,
+    onLoadedMetadata,
+    onPause,
+    onPlaying,
+  } = props;
+
+  const videoNodeRef = useRef();
+
+  useImperativeHandle(ref, () => ({
+    play: () => videoNodeRef.current.play(),
+    pause: () => videoNodeRef.current.pause(),
+    getMetadata: () => ({
+      title: title || '',
+      artist: artist || '',
+      album: album || '',
+      artwork: [{src: artwork || poster || ''}],
+    }),
+  }));
+
+  return (
+    <video
+      ref={videoNodeRef}
+      aria-label={ariaLabel}
+      src={src}
+      muted={muted}
+      loop={loop}
+      controls={controls}
+      onPlaying={onPlaying}
+      onPause={onPause}
+      poster={poster}
+      onLoadedMetadata={onLoadedMetadata}
+      onCanPlay={onLoad}
+      style={{position: 'relative', width: '100%', height: '100%'}}
```

Comment:
Do all provided videos have to have these styles? Is there any way we can do this for them?

Reply:
We could compose the style on top of the child styles, but that feels too intrusive and implicit. By making this explicit we also permit setting it some other way, like an author-defined classname.

Reply:
These are now being applied by the wrapper, allowing `<video>` drop-in without component adapter.